mono/metadata/cominterop.c

```diff
@@ -400,37 +401,34 @@ static MonoClass*
 cominterop_get_method_interface (MonoMethod* method)
 {
 	ERROR_DECL (error);
-	MonoClass *ic = method->klass;
+	MonoClass *ic;
+
+	if (MONO_CLASS_IS_INTERFACE_INTERNAL (method->klass))
```

Comment:
Is there anything substantial in the change to `cominterop_get_method_interface`? It looks like you just negated the logic and changed the loop to return directly.  Is that right?

Reply:
Whoops. The whole point was to return NULL here, which is also the answer to your other question. I'll fix that up and resend.

Anyway, in general this is just an attempt to simplify the code; I don't think it ended up being particularly relevant to the later patch implementing auto-dual interfaces (since that should never call compinterop_get_method_interface() with an interface).